### PR TITLE
Adding Initializers and Loss functions

### LINKED ISSUE #24: Add Initializers
Currently Nodes are defined to have spreads:

``` c++
            LinearNode(const int inputSize, const int outputSize,
                       float spread = 0.05,
                       const char *name="none") :
```

This needs to be refactored to provide an initializations class. 
Examples of initializations are : 
1. Le-Cun Normal
2. Glorot Uniform

### LINKED ISSUE #34: Loss functions
- [x] mean squared error
- [x] mean absolute error
- [x] binary cross entropy
- [ ] negative log likelyhood

### PR DESCRIPTION
**Pull request by pavanky**
- Fixes #24 
- Partially Fixes #34

### CODE REVIEW

**Comment by pavanky** [comment]
@umar456 @plavin @fohx13 This PR is done for now. Negative Log Likelihood will require adding lookup function to this repo so I'll look into this later.

Please review and merge when ready.

**Review by umar456 — APPROVED** [review]
